src/S3.jl

```diff
@@ -2,45 +2,29 @@ __precompile__()
 module S3
 
 using AWSCore
-using AWSS3
+using AWSS3: AWSS3
 using Dates
 using FilePathsBase
-using Memento
-
-import FilePathsBase: sync
 
 export S3Path, sync, upload
 
-const logger = getlogger(@__MODULE__)
-
-# Register the module level logger at runtime so that folks can access the logger via
-# `getlogger(MyModule)`. If this line is not included then the precompiled
-# `MyModule.logger` won't be registered at runtime.
-function __init__()
-    Memento.register(logger)
-end
-
-# Couple extra methods that should probably be included in AWSS3 at some point.
-"""
-    upload(src::AbstractPath, dest::S3Path)
+@deprecate S3Path(args...) AWSS3.S3Path(args...)
+@deprecate upload(src::AbstractPath, dest::AWSS3.S3Path) Base.cp(src, dest)
 
-Uploads a local file to the s3 path specified by `dest`.
-"""
-upload
-
-const upload = Base.cp
-
-# Note: naming copied from Go SDK:
-# https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/s3-example-presigned-urls.html
-function presign(
-    path::S3Path,
-    duration::Period=Hour(1);
-    config::AWSConfig=aws_config(),
+@deprecate(
+    presign(path::AWSS3.S3Path, duration::Period=Hour(1); config::AWSConfig=aws_config()),
```

Comment:
Reviewdog complains about the semicolon here but it is required.

Reply:
Yeah just ignore this! Weird. Maybe log an issue w/ [JuliaFormatter](https://github.com/domluna/JuliaFormatter.jl)?

Reply:
https://github.com/domluna/JuliaFormatter.jl/issues/475